aQute.libg/test/aQute/libg/command/CommandTest.java

```diff
@@ -4,22 +4,24 @@
 
 import org.junit.jupiter.api.Test;
 
+import aQute.lib.io.IO;
+
 public class CommandTest {
 
 	@Test
 	public void testSimple() throws Exception {
 		Command c = new Command();
-		c.add("java");
+		c.add(IO.getJavaExecutablePath("java"));
 		c.add("-Dx=\"foo bar\"");
 		c.add("-version");
-		assertThat(c.getArguments()).containsExactly("java", "-Dx=\"foo bar\"", "-version");
+		assertThat(c.getArguments()).containsExactly(IO.getJavaExecutablePath("java"), "-Dx=\"foo bar\"", "-version");
 		assertThat(c.execute(System.out, System.err)).isZero();
 	}
 
 	@Test
 	public void testFull() throws Exception {
-		Command c = new Command("java -Dx=\"foo bar\" -version");
-		assertThat(c.getArguments()).containsExactly("java", "-Dx=\"foo bar\"", "-version");
+		Command c = new Command(IO.getJavaExecutablePath("java") + " -Dx=\"foo bar\" -version");
```

Comment:
I just realized that this will fail if the absolute path string contains a space since the full string is parsed by spaces (This is why no one should really use the full command string). So for this test method, you will need to revert to the prior state (using simple `java` command) and delete the final assert on the execute method. We have confirmed the execution works in `testSimple` and this method confirms the full argument parsing, so we can avoid the assert on the execute here.

Reply:
next try. thank you for review